src/code/z_fbdemo_fade.c

```diff
@@ -1,17 +1,144 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_Start.s")
+#define THIS ((TransitionFade*)thisx)
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_Init.s")
+typedef enum {
+    /* 0 */ TRANS_FADE_DIR_IN,
+    /* 1 */ TRANS_FADE_DIR_OUT
+} TransitionFadeDirection;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_Destroy.s")
+typedef enum {
+    /* 0 */ TRANS_FADE_TYPE_NONE,
+    /* 1 */ TRANS_FADE_TYPE_ONE_WAY,
+    /* 2 */ TRANS_FADE_TYPE_FLASH
+} TransitionFadeType;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_Update.s")
+static Gfx sTransFadeSetupDL[] = {
+    gsDPPipeSync(),
+    gsSPClearGeometryMode(G_ZBUFFER | G_SHADE | G_CULL_BOTH | G_FOG | G_LIGHTING | G_TEXTURE_GEN |
+                          G_TEXTURE_GEN_LINEAR | G_LOD | G_SHADING_SMOOTH),
+    gsDPSetOtherMode(G_AD_NOTPATTERN | G_CD_BAYER | G_CK_NONE | G_TC_FILT | G_TF_BILERP | G_TT_NONE | G_TL_TILE |
+                         G_TD_CLAMP | G_TP_NONE | G_CYC_1CYCLE | G_PM_1PRIMITIVE,
+                     G_AC_NONE | G_ZS_PIXEL | G_RM_CLD_SURF | G_RM_CLD_SURF2),
+    gsDPSetCombineLERP(0, 0, 0, PRIMITIVE, 0, 0, 0, PRIMITIVE, 0, 0, 0, PRIMITIVE, 0, 0, 0, PRIMITIVE),
+    gsSPEndDisplayList(),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_Draw.s")
+TransitionInit TransitionFade_InitVars = {
+    TransitionFade_Init,   TransitionFade_Destroy, TransitionFade_Update,   TransitionFade_Draw,
+    TransitionFade_Start,  TransitionFade_SetType, TransitionFade_SetColor, NULL,
+    TransitionFade_IsDone,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_IsDone.s")
+void TransitionFade_Start(void* thisx) {
+    TransitionFade* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_SetColor.s")
+    switch (this->type) {
+        case TRANS_FADE_TYPE_NONE:
+            break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_fade/TransitionFade_SetType.s")
+        case TRANS_FADE_TYPE_ONE_WAY:
+            this->timer = 0;
+            this->color.a = (this->direction != TRANS_FADE_DIR_IN) ? 255 : 0;
+            break;
+
+        case TRANS_FADE_TYPE_FLASH:
+            this->color.a = 0;
+            break;
+    }
+    this->isDone = false;
```

Comment:
Could you add a default without breaking matching? (applies to other switch-cases in this file)